### ISSUE TITLE
python3 crash related to stdin being redirected on Windows

### ISSUE THREAD

**Issue by xaizek** [vim/vim#7053]
gVim started with `NUL` for standard input is aborted by Python3 on Windows.

**To Reproduce**

Only know how to reproduce it using [Vifm 0.11](https://vifm.info/news/2020-09-24.shtml) ([source](https://github.com/vifm/vifm/issues/592#issuecomment-701203820)):

1. Execute `:!gvim &`  in vifm
2. Execute `:pythonx print('test')` in gvim
3. Vim closes at this point.

**Expected behavior**

`test` printed on Vim's statusbar.

**Error message**

```
Fatal Python error: init_sys_streams: can't initialize sys standard streams
Python runtime state: core initialized    
OSError: [WinError 6] The handle is invalid      
Current thread 0x000025b0 (most recent call first):
<no Python frame> 
```

**Environment**
 - Vim v8.2.1381 or later
 - OS: Windows 7 x64, Windows 10
 - Python: 3.8.6
 - Terminal: GUI

**Additional context**

56b8dc331dc7f499c8221ed7d1c7795ad5e3bdeb from #6641 seems to be the cause of this issue. Vim v8.2.1377 works fine.

Description of #6641 says:

> Python 3.5 or later calls `io.py` in `Py_Initialize()`, and it calls
> `abort()` when stdin is redirected. This can be avoided by reconnecting
> stdin to CONIN$

However, I'm not sure that's the case, because I'm able to run `:!python &` in Vifm without hitting this error. And started this way Python's `stdin` is redirected to `NUL` pseudo-file.

My guess would be that reopening `CONIN$` fails when Vim is started with `stdin` redirected to `NUL`, which results in `stdin`  becoming invalid.

**Comment by k-takata** [comment]
Could you try #7061?

**Comment by bartlibert** [comment]
I can still reproduce this in gvim 8.2 with patch 1-1807. Should I create a new issue or can this one be reopened?

**Comment by k-takata** [comment]
I created a PR to fix this: #7106.

**Comment by xaizek** [comment]
c6ed254 (#7106) fixed this second time.
